engine/storage/datamotion/src/main/java/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -1685,13 +1777,17 @@ private CopyCmdAnswer copyImageToVolume(DataObject srcDataObject, VolumeInfo des
      * resign the SR and the VDI that should be inside of the snapshot before copying the VHD file to secondary storage.
      */
     private void createVolumeFromSnapshot(SnapshotInfo snapshotInfo) {
-        SnapshotDetailsVO snapshotDetails = handleSnapshotDetails(snapshotInfo.getId(), "create");
-
-        try {
-            snapshotInfo.getDataStore().getDriver().createAsync(snapshotInfo.getDataStore(), snapshotInfo, null);
-        }
-        finally {
-            _snapshotDetailsDao.remove(snapshotDetails.getId());
+        if ("true".equalsIgnoreCase(snapshotInfo.getDataStore().getDriver().getCapabilities().get("CAN_CREATE_TEMP_VOLUME_FROM_SNAPSHOT"))) {
+            prepTempVolumeForCopyFromSnapshot(snapshotInfo);
+        // fallback to original behavior of sending magic ddata to the create function
```

Comment:
can add return here without adding else statement

Reply:
Updated in upcoming commit.